### ISSUE TITLE
Dark ISE Theme

### ISSUE THREAD

**Issue by mattmcnabb** [PowerShell/vscode-powershell#1051]
So I'm thinking about adding a dark theme to go along with the ISE theme added this spring. My thoughts are that I'd like to keep the console colors as-is, but have dark theming in the editor. Also, I haven't decided whether the UI theming should be kept light like the current PowerShell ISE theme, or whether it should follow the editor theme.

Also, before I begin I'd like to get others thoughts on what they would like in a dark theme for the PowerShell extension. Some of the popular dark themes from other editors are Solarized, Monokai, Dracula,etc. Also, I just found a theme for VSCode from Cometeer called Spacemacs which is quite nice. If you're interested in a dark theme for PowerShell, please reply here with ideas for how you'd like it to look.

**Comment by dotps1** [comment]
dark theme that came in the spring?  is that with the ISE preview thats in the gallery?

**Comment by mattmcnabb** [comment]
@dotps1 This spring we added the PowerShell ISE Theme to the VSCode Extension. This theme mimics the color style of the ISE's defaults. The thought was that it would make transition to VSCode a bit more comfortable for those used to the ISE. It comes with the bright white editor and blue console that we all know and love. This was made possible with VSCode's 1.13 release which added the capability to color the terminal separately from the editor.

I'd like to improve on that by adding another theme that keeps the integrated console's blue coloring, but with dark tones in the editor.

**Comment by daviwil** [comment]
A dark theme would be awesome!  I'm OK with whatever direction you want to go with it, maybe start working on something that looks good to you and then post screenshots here for feedback from others?

**Comment by pcgeek86** [comment]
Have you seen the Blackboard theme? I'm not really sure why, but it looks really pleasant and nice. I think it's the contrast that makes it easy to read, without being overly saturated color-wise.

![image](https://user-images.githubusercontent.com/466713/31595751-b9ee0268-b1f2-11e7-9fed-7121f3ed4f83.png)

**Comment by mattmcnabb** [comment]
Working on a solarized one, but still have some UI elements to theme

![image](https://user-images.githubusercontent.com/6998507/31643459-a98a3d48-b2be-11e7-8567-73c956c9b568.png)

Maybe we should have more than one?

**Comment by tylerl0706** [comment]
context: This should be just modifying https://github.com/PowerShell/vscode-powershell/blob/master/themes/theme-psise/theme.json

as far as I know.

**Comment by mbrinkho** [comment]
Just a thought - if you change the user settings you can get the white on blue with any theme you like.

```
    "workbench.colorCustomizations": {
        "terminal.background": "#012456",
        "terminal.foreground": "#F5F5F5"    }
```

**Comment by mattmcnabb** [comment]
I know it's been a while, but I just got around to looking at this again. Any objections if I start with recreating the default themes, but with the PowerShell console colors?

![powershell dark visual studio](https://user-images.githubusercontent.com/6998507/47333129-c921dc00-d64f-11e8-9fb0-0b77c1efc9e3.png)
![powershell dark](https://user-images.githubusercontent.com/6998507/47333130-c921dc00-d64f-11e8-90f2-999ece1069f2.png)
![powershell light visual studio](https://user-images.githubusercontent.com/6998507/47333131-c921dc00-d64f-11e8-8ea8-db33f709123e.png)
![powershell light](https://user-images.githubusercontent.com/6998507/47333132-c921dc00-d64f-11e8-9090-43e84b6068af.png)
![powershell quiet light](https://user-images.githubusercontent.com/6998507/47333133-c921dc00-d64f-11e8-8746-9fbb00423760.png)

**Comment by mattmcnabb** [comment]
@tylerl0706 Thanks! I'll get this done soon - just want to add a little pizzazz to the terminal title bar in dark themes. What branch should I submit the PR to?

**Comment by tylerl0706** [comment]
Master is fine :)

**Comment by westerdaled** [comment]
Did the Dark ISE Theme or Solarized Dark ISE  ever get released int the wild to be easily installed within VSCode ?   In my case I don't want to amend any of my currently installed themes just to have Dark ISE or Solarized Dark ISE when I toggle ISE Mode.

**Comment by SydneyhSmith** [comment]
@westerdaled nope Dark ISE mode is still up for grabs!

**Comment by ZenoArrow** [comment]
I know it sounds dumb/ungrateful, but the lack of a dark theme is the one thing putting me off using the ISE mode (I'm a fan of the idea of the ISE mode, but the appearance of Powershell ISE was never its strong point, its main strengths were its simplicity and convenience). Is there a way to hack the colours used without having to create an extension (I don't mind if updates break these hacks)? I don't have the inclination to jump through the hoops of creating and maintaining an extension, I just want to be able to use the ISE mode without having to fry my eyesight with the current UI.

**Comment by TylerLeonhardt** [comment]
You can use ISE Mode with a different theme.

All you have to do is ctrl+shift+p to open the command pallet and set the color theme to whatever you'd like.

One day, we may get an actual Dark ISE theme or a PowerShell blue console theme that you can apply on any theme... But we don't have it yet. When that comes, we can make ISE mode be configurable.

**Comment by ZenoArrow** [comment]
@TylerLeonhardt , thank you for the advice.

**Comment by Jess-S13** [comment]
Forgive me if i'm asking this in the wrong forum, but is there a way in ISE-Mode to select the "Dark Editor"/"Dark Console" Theme that was used in PowerShell ISE? I tried just scrolling thru the massive number of themes in vscode and am having an issue finding this? 
![image](https://user-images.githubusercontent.com/105818693/169140550-0982fbe1-1d25-4e05-8046-131c34853eff.png)

Thank you in advance,
Jess

**Comment by pcgeek86** [comment]
@Jess-S13 This GitHub repository is for the PowerShell extension for Microsoft Visual Studio Code. The ISE is pretty much dead; I'd highly recommend getting on board with VSCode. It's buggy, and far from perfect, but there are lots of benefits to learning about VSCode and altering your workflow somewhat. Believe me. 😁

**Comment by panekj** [comment]
@pcgeek86 They are asking about VSCode themes, not ISE.

**Comment by Jess-S13** [comment]
> @Jess-S13 This GitHub repository is for the PowerShell extension for Microsoft Visual Studio Code. The ISE is pretty much dead; I'd highly recommend getting on board with VSCode. It's buggy, and far from perfect, but there are lots of benefits to learning about VSCode and altering your workflow somewhat. Believe me. 😁

Hi @pcgeek86,
My apologies I was not clear. Im migrating from ISE to VSCode. Im trying to find out if the ISE-Mode in VSCode has the ISE Theme of "Dark Console"/"Dark Editor" which grabbed the screenshot of to show what I was referrring to. The ISE Mode light theme works fine but is very bright. 

Thank you for responding though!

**Comment by pcgeek86** [comment]
@Jess-S13 I totally missed what you meant, sorry about that!

Can you just use a different dark theme, or are you specifically trying to replicate a very specific dark theme from the ISE? There are *tons* of theme packs that offer a variety of alternative themes. 

If I'm totally off-base, sorry again! 😆

**Comment by pcgeek86** [comment]
I use the `Outrun Night` theme for VSCode. Here's what my setup looks like:

https://marketplace.visualstudio.com/items?itemName=samrapdev.outrun

Extension URL: ![image](https://user-images.githubusercontent.com/466713/169142642-93b68206-fda2-4b9f-9d49-7939129c4f2d.png)

**Comment by Jess-S13** [comment]
Hey @pcgeek86,
No apology needed, im just thrilled someone is answering. Ive only ever worked in Powershell ISE as my team has only ever used 5x, but now were going to expand some use cases to 7x so im making the jump. The frustration I think is the best way of stating it, im experiencing is how the themes do not handle the different verbage in powershell the same way, an example being the amazing looking "Azure Contrast" here in this screenshot, colors strings & variables the same and variables and options the same, which is a large shift from ISE. 
![image](https://user-images.githubusercontent.com/105818693/169143479-a1b5efa6-08ea-464f-913b-3e1c79859348.png)
Actual PowerShell ISE:
![image](https://user-images.githubusercontent.com/105818693/169143528-80157d20-b90e-44f1-86c3-bb1a18c4cbf0.png)
ISE Mode in VSCode does this right (for the most part), so im thinking its a function of the theme, but as im just making the jump i was hoping it may be in the ISE mode itself as an option:
![image](https://user-images.githubusercontent.com/105818693/169143805-1bb692cc-843c-4045-8c3c-ef5f791d685c.png)

**Comment by pcgeek86** [comment]
@Jess-S13 

I'll be totally honest. I'm not even sure what ISE Mode in Visual Studio Code is. 😆 

To make sure I'm understanding the differences that you're seeing properly .... are you expecting tokens like `MyFunction` and `Parameter` and `Get-ChildItem` to all have different colors? And your team is concerned because they don't look quite as different in VSCode?

You're right that the theme would control colorization of all of the PowerShell language tokens. I'd recommend playing around with a few different themes out on the VSCode Extensions Marketplace, and see how you like them. I don't think you'll ever get an exact one-to-one replication between the colors in ISE and VSCode, unless you invest a lot of engineering effort.

Each of your team members would probably enjoy having the flexibility of swapping their theme on a daily, weekly, or monthly basis! It's much easier in VSCode than ISE.

Personally, I think your [team's] time would be best spent accepting that there will be some minor colorization differences in VSCode, and focus on developing the best code possible. 🎉 There will be significant changes in how VSCode and ISE behave. The net benefit of using VSCode is 100% worth it though!

Again, apologies if I'm misinterpreting something about your question!

**tl;dr** Play around with a bunch of VSCode themes and see which ones each of you like the best. Embrace the thematic differences! Some are more contrasty, others less so.

**Comment by Jess-S13** [comment]
Hi @pcgeek86,
From what im gathering from https://docs.microsoft.com/en-us/powershell/scripting/dev-cross-plat/vscode/how-to-replicate-the-ise-experience-in-vscode?view=powershell-7.2 doc the ISE-Mode is part of the powershell extension which carries over alot of functionality from PowerShell ISE to VSCode to simplify the migration process for new users like myself.  One of the parts of the ISE-Mode is the "ISE-Mode" theme which lines up really close to the PowerShell ISE Dark Console/Light Editor theme from Powershell-ISE. As Ive always used the Dark/Dark theme and I saw this thread I figured it was worth asking if this already exists and I just missed it before I try to look up my own.

**Comment by Jess-S13** [comment]
For those who may stumble upon this after me. I followed this page -> https://www.alveeakand.com/how-to-modify-themes-in-vscode/ to edit the "ISE-Mode" theme to match the "Dark Editor"/"Dark Console" theme from the original ISE. You need the add in the following to your settings.json file and it will make it match. I had to add in a bunch of json lines at the bottom so the settings page was readable, but basically makes anything not powershell simply white on blue.. Hope it helps for anyone else:

Here is how it looks:
![image](https://user-images.githubusercontent.com/105818693/169421092-cd2a9e3c-8fbf-4170-ae7c-58304a705864.png)


`"workbench.activityBar.visible": false,
    "debug.openDebug": "neverOpen",
    "workbench.startupEditor": "none",
    "workbench.colorCustomizations": {
        "[PowerShell ISE]": {            
            "terminal.foreground" : "#EDEEF0",
            "terminal.background" : "#012456",
            "editor.background": "#012456",
            "editorCursor.foreground": "#f5f5f5",
            "editor.lineHighlightBackground": "#012456",
        }
    },
    "editor.tokenColorCustomizations": {
        "[PowerShell ISE]": {
            "textMateRules": [
                {
                    "scope": [
                        "comment.line.powershell",
                    ],
                    "settings": {
                        "foreground": "#98fb98"
                    }
                },
                {
                    "scope": [
                        "support.function.attribute.powershell",
                    ],
                    "settings": {
                        "foreground": "#b0c4de"
                    }
                },
                {
                    "scope": [
                        "support.function.powershell",
                        "meta.function.powershell",
                        "meta.scriptblock.powershell",
                        "keyword.control.powershell",
                    ],
                    "settings": {
                        "foreground": "#e0ffff"
                    }
                },
                {
                    "scope": [
                        "entity.name.function.powershell",
                        "source.powershell",
                    ],
                    "settings": {
                        "foreground": "#ee82ee"
                    }
                },
                {
                    "scope": [
                        "punctuation.section.group.end.powershell",
                        "punctuation.section.group.begin.powershell",
                        "keyword.operator.other.powershell",
                        "variable.parameter.attribute.powershell",
                        "punctuation.terminator.statement.powershell",
                        "punctuation.section.braces.begin.powershell",
                        "punctuation.section.braces.end.powershell",
                        "meta.group.simple.subexpression.powershell",
                        "variable.other.member.powershell",
                        "support.type.property-name.json",
                        "string.json",
                        "punctuation.definition.dictionary.end.json",
                        "punctuation.definition.dictionary.begin.json",
                        "punctuation.separator.dictionary.pair.json",
                        "punctuation.separator.array.json",
                        "punctuation.definition.array.begin.json",
                        "punctuation.definition.array.end.json",
                        "punctuation.separator.dictionary.key-value.json",
                        "support.type.property-name.json",
                        "punctuation.support.type.property-name.begin.json",
                        "punctuation.support.type.property-name.end.json",
                        "string.quoted.double.json",
                        "punctuation.definition.string.begin.json",
                        "punctuation.definition.string.end.json",
                    ],
                    "settings": {
                        "foreground": "#f5f5f5"
                    }
                },
                {
                    "scope": [
                        "constant.numeric.integer.powershell",
                    ],
                    "settings": {
                        "foreground": "#ffe4c4"
                    }
                },
                {
                    "scope": [
                        "punctuation.section.bracket.begin.powershell",
                        "punctuation.section.bracket.end.powershell",
                    ],
                    "settings": {
                        "foreground": "#d3d3d3"
                    }
                },
                {
                    "scope": [
                        "string.quoted.single.powershell",
                        "string.quoted.double.powershell",
                        "punctuation.definition.string.begin.powershell",
                        "punctuation.definition.string.end.powershell",
                    ],
                    "settings": {
                        "foreground": "#db7094"
                    }
                },
                {
                    "scope": [
                        "variable.other.readwrite.powershell",
                        "constant.language.powershell",
                    ],
                    "settings": {
                        "foreground": "#ff4400"
                    }
                },
                {
                    "scope": [
                        "storage.type.powershell",
                    ],
                    "settings": {
                        "foreground": "#e0ffff"
                    }
                },
            ],
        }
    },
    "editor.tabCompletion": "on",
    "editor.autoClosingBrackets": "never",
    "editor.autoClosingQuotes": "never",
    "editor.defaultFormatter": "ms-vscode.powershell",
    "powershell.integratedConsole.focusConsoleOnExecute": false,
    "editor.wordSeparators": "`~!@#%^&*()-=+[{]}\\|;:'\",.<>/?",
    "powershell.buttons.showPanelMovementButtons": true,
    "workbench.colorTheme": "PowerShell ISE",
    "editor.bracketPairColorization.enabled": false`
